pkg/server/cache/entrycache/fullcache.go

```diff
@@ -173,6 +178,37 @@ func Build(ctx context.Context, entryIter EntryIterator, agentIter AgentIterator
 	}, nil
 }
 
+func Update(ctx context.Context, ds datastore.DataStore, cache *FullEntryCache) error {
+	req := &datastore.ListEventsRequest{
+		LastID: cache.lastEventID,
+	}
+	resp, err := ds.ListEvents(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	for _, entryID := range resp.EntryIDs {
+		commonEntry, err := ds.FetchRegistrationEntry(ctx, entryID)
+		if err != nil {
+			return err
+		}
+
+		if commonEntry == nil {
+			cache.deleteEntry(entryID)
+			cache.lastEventID++
+			continue
+		}
+
+		if err := cache.createOrUpdateEntry(commonEntry); err != nil {
+			return err
+		}
+
+		cache.lastEventID++
```

Comment:
I agree, instead of incrementing, the datastore should return the event ID along with the entry ID and that should be used to march this forward. That feels more robust.

Reply:
How should we handle the case of an error in `createOrUpdateEntry`? in that case we will have lost the event.